src/http/matcher/mod.rs

```diff
@@ -35,18 +35,21 @@ use crate::{
     stream::matcher::SocketMatcher,
 };
 
+use core::fmt;
```

Comment:
imports should be on top, and you can just import `std::fmt`. Given it's a framework for proxies I do not see a reason at the moment to try to make it work non-std.

Once we are more stable I can revisit that POV, but for now importing from `std` instead of `core` is what shall be done please.